core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -720,6 +720,12 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] with Serializable { self =>
   def short(index: Int)(implicit ev: A <:< Short): Short =
     ev(apply(index))
 
+  override def slice(from: Int, until: Int): Chunk[A] = {
+    val start = if (from < 0) 0 else if (from > length) length else from
+    val end   = if (until < start) start else if (until > length) length else until
+    Chunk.Slice(self, start, end - start)
```

Comment:
Not strictly required but it can be a good idea to avoid creating the slice if from = 0, until = end.

Reply:
Interestingly this results in about a 20% reduction in the throughput of the stream benchmark, I think due to additional branching that doesn't get optimized. Of course I'm sure there are other cases where you don't actually need to create the slice that would benefit from this.

Reply:
All right, let's skip it. I was mostly thinking of stack safety in cases where slicing is done recursively based on dynamically computed bounds.